### PR TITLE
Search

### PR DESCRIPTION
**Pull request by cassiemak**
- Search with the 4 params
- added loading image
- cleaned up code
- factored sliders directives
- linked the slider to the data = params
- no search result image

### CODE REVIEW

**Comment by h8rry** [comment]
`app/assets/javascripts/angular/controllers/home_controller.js.erb` has too much code. You need to break in it down into services.

File: bower.json [hunk 1]
```diff
@@ -8,6 +8,6 @@
     "underscore": "~1.4.4",
     "angular-ui-utils": "0.2.1",
     "angular-ui": "0.4.0",
-    "angular-google-maps": "2.0.12"
+    "angular-google-maps": "2.0.12"
```
Comment (h8rry) [inline bower.json:11]: undo the empty space please

File: app/models/region_trail_link.rb [hunk 2]
```diff
@@ -1,4 +0,0 @@
-class RegionTrailLink < ActiveRecord::Base
```
Comment (h8rry) [inline app/models/region_trail_link.rb:1]: @cchchoi1986 Take a look at this. Trail and Region have one-to-many relationship, not many-to-many, so we changed it.

File: app/models/region.rb [hunk 3]
```diff
@@ -1,8 +1,7 @@
 class Region < ActiveRecord::Base
   has_many :region_bird_links
   has_many :birds, through: :region_bird_links
-  has_many :region_trail_links
-  has_many :trails, through: :region_trail_links
+  has_many :trails
```
Comment (h8rry) [inline app/models/region.rb:4]: @cchchoi1986 Take a look at this. Trail and Region have one-to-many relationship, not many-to-many, so we changed it.

File: app/views/trails/show.json.jbuilder [hunk 4]
```diff
@@ -8,4 +8,11 @@ json.distance @trail.distance
 json.duration @trail.duration
 json.food_supply @trail.food_supply
 json.warning @trail.warning
-json.photo_urls @trail.photo_urls
\ No newline at end of file
+json.photo_urls @trail.photo_urls
+json.start_longitude @trail.start_longitude
```
Comment (cchchoi1986) [inline app/views/trails/show.json.jbuilder:12]: these should be 
json.start_coordinates @trail.start_coordinates
json.end_coordinates @trail.end_coordinates

File: app/models/region_trail_link.rb (hunk 2 above)
Comment (cchchoi1986) [inline app/models/region_trail_link.rb:1]: agree

File: db/migrate/20150215074541_create_region_trail_links.rb [hunk 5]
```diff
@@ -1,9 +0,0 @@
-class CreateRegionTrailLinks < ActiveRecord::Migration
-  def change
-    create_table :region_trail_links do |t|
-      t.integer :region_id
```
Comment (cchchoi1986) [inline db/migrate/20150215074541_create_region_trail_links.rb:4]: agree

File: app/views/trails/show.json.jbuilder [hunk 6]
```diff
@@ -8,4 +8,11 @@ json.distance @trail.distance
...
+json.start_latitude @trail.start_latitude
+json.end_longitude @trail.end_longitude
+json.end_latitude @trail.end_latitude
+json.region_id @trail.region_id
+json.birds @trail.region.birds
```
Comment (cchchoi1986) [inline app/views/trails/show.json.jbuilder:17]: if we do this we can remove the show.json.jbuilder + index.json.jbuilder on plants and birds views.  also their controllers are not used anymore.  I can fix after merge

File: app/views/trails/show.json.jbuilder (hunk 4 above)
Comment (cassiemak) [inline app/views/trails/show.json.jbuilder:12]: I looked at the rails console, has it been changed?
If so, I will amend the variables...

File: db/migrate/20150215074541_create_region_trail_links.rb (hunk 5 above)
Comment (cassiemak) [inline db/migrate/20150215074541_create_region_trail_links.rb:4]: Somehow we need to change the scrape/ rake file to add region Id to the
trails??

File: app/views/trails/show.json.jbuilder (hunk 6 above)
Comment (cassiemak) [inline app/views/trails/show.json.jbuilder:17]: Cool!

**Pull request merged**